likeminds-chat-reactnative-integration/ChatSX/customModals/ChatroomActionModal/index.tsx

```diff
@@ -0,0 +1,101 @@
+  const {
+    chatroomID,
+    chatroomType,
+    chatroomDBDetails,
+    navigation,
+    isSecret,
+    filteredChatroomActions,
+    modalVisible,
+    setModalVisible,
+    handleModalClose,
+    showWarningModal,
+    joinChatroom,
+    muteNotifications,
+    unmuteNotifications,
+    handleBlockMember,
+    unblockMember,
+  }: ChatroomContextValues = useChatroomContext();
+  return (
+    <Modal
+      transparent={true}
+      visible={modalVisible}
+      onRequestClose={() => {
+        setModalVisible(!modalVisible);
+      }}
+    >
+      <Pressable style={styles.centeredView} onPress={handleModalClose}>
+        <View>
+          <Pressable onPress={() => {}} style={[styles.modalView]}>
+            {filteredChatroomActions?.map((val: any, index: any) => {
+              return (
+                <TouchableOpacity
+                  onPress={async () => {
+                    if (val?.id === ChatroomActions.VIEW_PARTICIPANTS) {
+                      setModalVisible(false);
+                      // navigation.navigate(VIEW_PARTICIPANTS, {
+                      //   chatroomID: chatroomID,
+                      //   isSecret: isSecret,
+                      //   chatroomName: chatroomName,
+                      // });
+                    } else if (
+                      val?.id === ChatroomActions.LEAVE_CHATROOM ||
+                      val?.id === ChatroomActions.LEAVE_SECRET_CHATROOM
+                    ) {
+                      // showWarningModal();
+                      setModalVisible(false);
+                    } else if (val?.id === ChatroomActions.JOIN_CHATROOM) {
+                      if (!isSecret) {
+                        // joinChatroom();
+                      }
+                      setModalVisible(false);
+                    } else if (val?.id === ChatroomActions.MUTE_NOTIFICATIONS) {
+                      // await muteNotifications();
+                      setModalVisible(false);
+                    } else if (
+                      val?.id === ChatroomActions.UNMUTE_NOTIFICATIONS
+                    ) {
+                      // await unmuteNotifications();
+                      setModalVisible(false);
+                    } else if (val?.id === ChatroomActions.VIEW_PROFILE) {
+                      //View Profile code
+                    } else if (val?.id === ChatroomActions.BLOCK_MEMBER) {
+                      // await handleBlockMember();
+                      setModalVisible(false);
+                    } else if (val?.id === ChatroomActions.UNBLOCK_MEMBER) {
+                      // await unblockMember();
+                      setModalVisible(false);
+                    } else if (val?.id === ChatroomActions.SHARE) {
+                      //   Share flow
+                      //   onShare(
+                      //     chatroomID,
+                      //     chatroomType,
+                      //     chatroomDBDetails?.isSecret
+                      //   );
```

Comment:
not understood why muteNotifications(), unblockMember(), onShare() etc are commented out